flatson/flatson.py

```diff
@@ -137,3 +124,20 @@ def flatten_dict(self, obj):
         """Return an OrderedDict dict preserving order of keys in fieldnames
         """
         return OrderedDict(zip(self.fieldnames, self.flatten(obj)))
+
+    def infer_flattened_field_names(self, schema, field_sep='.'):
```

Comment:
Interesting refactoring!
I was refraining to make this a method because it recurs.
I usually avoid making recursive methods because I feel they invite adding complexity, I prefer any recursive stuff to be pure functions (that is, independent of external state like `self.ignore`).
But I may be being too conservative. :)

This being recursive, it's interesting to think about what we want this `ignore` global option to mean. Does it mean any key with that name should be ignored, even if it's in a nested object (the current behavior)? Or should it mean only keys of the upper-level object? We should also have test for this situation.

One other thing, if we keep this as a method, it's good to rethink its arguments -- e.g. `field_sep` can be discarded and we can use `self.field_sep` directly.